components/use-invoice.js

```diff
@@ -16,24 +18,29 @@ export default function useInvoice () {
       throw error
     }
 
-    const { cancelled, cancelledAt, actionError, expiresAt, forwardStatus } = data.invoice
+    const { cancelled, cancelledAt, actionState, actionError, expiresAt, forwardStatus } = data.invoice
 
     const expired = cancelledAt && new Date(expiresAt) < new Date(cancelledAt)
     if (expired) {
       throw new InvoiceExpiredError(data.invoice)
     }
 
-    const failed = cancelled || actionError
-
-    if (failed && (forwardStatus && forwardStatus !== 'CONFIRMED')) {
+    const failedForward = forwardStatus && forwardStatus !== 'CONFIRMED'
+    if (failedForward) {
```

Comment:
Removing `failed` from the condition for failed forwards makes this resolve a little faster since the withdrawal can already have failed before the invoice itself is marked as failed, see video in https://github.com/stackernews/stacker.news/issues/1707#issuecomment-2535829482.